File: code/phylogenies/gtdbtk-v-pepm-tree-comparisons/treecompare.R
library(tidyverse)
library(here)
library(ape)
library(treeman)
library(dendextend)
library(DECIPHER)

#### load trees
genome_bac <- read.tree(file=here("data", "gtdbtk.bac120.user_msa.trimgappyout.shared.tre"))
pepm_bac <- read.tree(file=here("data", "bacteria_PEPM.curated.gappyout.genomes.shared.tre"))

genome_arc <- read.tree(file=here("data", "gtdbtk.ar122.user_msa.trimgappyout.shared.tre"))
pepm_arc <- read.tree(file=here("data", "archaea_PEPM.curated.gappyout.genomes.shared.tre"))

random_bac <- rtree(length(genome_bac$tip.label), tip.label = genome_bac$tip.label, rooted=FALSE)
random_arc <- rtree(length(genome_arc$tip.label), tip.label = genome_arc$tip.label, rooted=FALSE)


ete3 compare -t randomtree.tre -r pruned_species.tre --unrooted


## calculate distances between trees for bacteria
dist.topo(genome_bac, genome_bac, method = "PH85")
dist.topo(genome_bac, pepm_bac, method = "PH85")
dist.topo(genome_bac, random_bac, method = "PH85")

dist.topo(genome_bac, genome_bac, method = "score")
dist.topo(genome_bac, pepm_bac, method = "score")
dist.topo(genome_bac, random_bac, method = "score")

association <- cbind(genome_bac$tip.label, pepm_bac$tip.label)
cophyloplot(genome_bac, pepm_bac, assoc = association,
            length.line = 5, space = 68, gap = 3, show.tip.label = FALSE)

a <- ReadDendrogram(here("data", "gtdbtk.bac120.user_msa.trimgappyout.shared.tre"))
b <- ReadDendrogram(here("data", "bacteria_PEPM.curated.gappyout.genomes.shared.tre"))

#### yuck...
tanglegram(a, b, 
           common_subtrees_color_lines=FALSE, 
           highlight_distinct_edges=FALSE, 
           highlight_branches_lwd=FALSE, 
           highlight_branches_col=FALSE, 
           remove_nodePar=TRUE,
           faster=TRUE)

## calculate distances between trees for archaea
dist.topo(genome_arc, genome_arc, method = "PH85")
dist.topo(genome_arc, pepm_arc, method = "PH85")
dist.topo(genome_arc, random_arc, method = "PH85")

dist.topo(genome_arc, genome_arc, method = "score")
dist.topo(genome_arc, pepm_arc, method = "score")
dist.topo(genome_arc, random_arc, method = "score")

a <- ReadDendrogram(here("data", "gtdbtk.ar122.user_msa.trimgappyout.shared.tre"))
b <- ReadDendrogram(here("data", "archaea_PEPM.curated.gappyout.genomes.shared.tre"))

tanglegram(a, b, 
           common_subtrees_color_lines=TRUE, 
           highlight_distinct_edges=TRUE, 
           highlight_branches_lwd=FALSE, 
           highlight_branches_col=FALSE, 
           remove_nodePar=FALSE,
           faster=TRUE)

File: code/phylogenies/gtdbtk-v-pepm-tree-comparisons/bin/tree_compare.py
#!/usr/bin/python
import pandas as pd
from ete3 import Tree

##### bacteria
genome_bac = Tree("gtdbtk.bac120.user_msa.trimgappyout.shared.tre")
pepm_bac = Tree("bacteria_PEPM.curated.gappyout.genomes.shared.tre")

bacnames = [leaf.name for leaf in genome_bac]
baclen = len(genome_bac)

bac_tree_d = {}
bac_tree_d['pepm'] = pepm_bac
bac_tree_d['self'] = genome_bac
for k in range(0, 10, 1):
	t=Tree()
	t.populate(baclen, bacnames)
	bac_tree_d["rtree_"+str(k)] = t

bac_result_d={}
for tree, obj in bac_tree_d.items():
	result = obj.compare(genome_bac, unrooted='True')
	bac_result_d[tree] = {
						'rf': result['rf'], 
						'max_rf': result['max_rf'],
						'norm_rf': result['norm_rf'],
						'effective_tree_size': result['effective_tree_size'],
						'ref_edges_in_source': result['ref_edges_in_source'],
						'source_edges_in_ref': result['source_edges_in_ref'],
						'source_subtrees': result['source_subtrees']
						}

##### archaea
genome_arc = Tree("gtdbtk.ar122.user_msa.trimgappyout.shared.tre")
pepm_arc = Tree("archaea_PEPM.curated.gappyout.genomes.shared.tre")

arcnames = [leaf.name for leaf in genome_arc]
arclen = len(genome_arc)

arc_tree_d = {}
arc_tree_d['pepm'] = pepm_arc
arc_tree_d['self'] = genome_arc
for k in range(0, 10, 1):
	t=Tree()
	t.populate(arclen, arcnames)
	arc_tree_d["rtree_"+str(k)] = t

arc_result_d={}
for tree, obj in arc_tree_d.items():
	result = obj.compare(genome_arc, unrooted='True')
	arc_result_d[tree] = {
						'rf': result['rf'], 
						'max_rf': result['max_rf'],
						'norm_rf': result['norm_rf'],
						'effective_tree_size': result['effective_tree_size'],
						'ref_edges_in_source': result['ref_edges_in_source'],
						'source_edges_in_ref': result['source_edges_in_ref'],
						'source_subtrees': result['source_subtrees']
						}

##### combine dicts
b = pd.DataFrame.from_dict(bac_result_d, orient='index')
b['group'] = "bacteria"

a = pd.DataFrame.from_dict(arc_result_d, orient='index')
a['group'] = "archaea"

final = pd.concat([b,a])
final.to_csv("tree_comparison.csv", header=True, index_label="tree")